src/middlewared/middlewared/plugins/service_/utils.py

```diff
@@ -0,0 +1,28 @@
+import enum
+from middlewared.utils.privilege import app_credential_full_admin_or_user
+
+class ServiceWriteRole(enum.Enum):
+    CIFS = 'SHARING_SMB_WRITE'
+    NFS = 'SHARING_NFS_WRITE'
+    ISCSITARGET = 'SHARING_ISCSI_WRITE'
+
+
+def app_has_write_privilege_for_service(
+    app: object | None,
+    service: str
+) -> bool:
+    if app_credential_full_admin_or_user(app, None):
```

Comment:
It's not immediately obvious to me what happens when the `username` (2nd arg to this function) is `None`.

Reply:
It just isn't evaluated. I refactored to use `credential_has_full_admin`